File: BookCircle.Core/Entities/Notification.cs
using BookCircle.Core.Enums;

namespace BookCircle.Core.Entities;

public class Notification
{
    public int Id { get; set; }
    public string UserId { get; set; } = string.Empty;
    public NotificationType Type { get; set; }
    public string Message { get; set; } = string.Empty;
    public bool IsRead { get; set; } = false;
    public int? RelatedEntityId { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public User User { get; set; } = null!;
}

File: bookcircle-Frontend/src/entities/notification.ts
export type NotificationType =
  | 'BorrowRequest'
  | 'BorrowAccepted'
  | 'BorrowRejected'
  | 'BookApproved'
  | 'BookRejected'
  | 'NewComment'

export interface Notification {
  id: number
  userId: string
  type: NotificationType
  message: string
  isRead: boolean
  createdAt: string
  relatedEntityId?: number
}
